Mathlib/SetTheory/Ordinal/Notation.lean

```diff
@@ -100,42 +100,37 @@ instance : Preorder ONote where
   le_trans _ _ _ := @le_trans Ordinal _ _ _ _
   lt_iff_le_not_le _ _ := @lt_iff_le_not_le Ordinal _ _ _
 
-theorem lt_def {x y : ONote} : x < y ↔ repr x < repr y :=
-  Iff.rfl
-
-theorem le_def {x y : ONote} : x ≤ y ↔ repr x ≤ repr y :=
-  Iff.rfl
+theorem lt_def {x y : ONote} : x < y ↔ repr x < repr y := Iff.rfl
+theorem le_def {x y : ONote} : x ≤ y ↔ repr x ≤ repr y := Iff.rfl
 
 instance : WellFoundedRelation ONote :=
   ⟨(· < ·), InvImage.wf repr Ordinal.lt_wf⟩
 
-/-- Convert a `Nat` into an ordinal -/
-@[coe]
-def ofNat : ℕ → ONote
+instance : NatCast ONote where
+  natCast
   | 0 => 0
   | Nat.succ n => oadd 0 n.succPNat 0
 
--- Porting note (#11467): during the port we marked these lemmas with `@[eqns]`
--- to emulate the old Lean 3 behaviour.
-
-@[simp] theorem ofNat_zero : ofNat 0 = 0 :=
-  rfl
+/-- Convert a natural number to an ordinal notation -/
+@[deprecated (since := "2024-10-17")]
+def ofNat (n : ℕ) : ONote := n
 
-@[simp] theorem ofNat_succ (n) : ofNat (Nat.succ n) = oadd 0 n.succPNat 0 :=
-  rfl
+@[simp] theorem natCast_zero : (0 : ℕ) = (0 : ONote) := rfl
+@[simp] theorem natCast_succ (n : ℕ) : n.succ = oadd 0 n.succPNat 0 := rfl
+theorem natCast_one : (1 : ℕ) = (1 : ONote) := rfl
 
-instance nat (n : ℕ) : OfNat ONote n where
-  ofNat := ofNat n
+instance (n : ℕ) : OfNat ONote n where
+  ofNat := n
```

Comment:
For these instances you should at least have a lemma `natCast_ofNat : natCast (noindex (OfNat.ofNat n)) = OfNat.ofNat n`

(assuming you don't get this for free with a later `AddCommMonoidWithOne` instance).